diff --git a/server/web/api/m3u.go b/server/web/api/m3u.go
--- a/server/web/api/m3u.go
+++ b/server/web/api/m3u.go
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+	"os"
 	"path/filepath"
 	"sort"
 	"strconv"
@@ -187,6 +188,10 @@ func sendM3U(c *gin.Context, name, hash string, m3u string) {
 }
 
 func getM3uList(tor *state.TorrentStatus, host string, fromLast bool, startIndex string) string {
+	if customHost, ok := os.LookupEnv("M3U_CUSTOM_HOST"); ok && customHost != "" {
+		host = strings.TrimRight(customHost, "/")
+	}
+
 	m3u := ""
 	from := 0
 	if startIndex != "" {
